### PR TITLE
[AOT Refactor] logging utils

### PR DESCRIPTION
**Pull request by jon-chuang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114643
* #114564
* #114563
* #114562
* #114561
* #114559
* #114558
* #114557
* #114556
* #114555
* #114554
* #114553
* #114552
* __->__ #114551
* #114550

---

Part _ of https://github.com/pytorch/pytorch/issues/114548

### CODE REVIEW

**Comment by jon-chuang** [comment]
I think merge failure is unrelated as all other PRs in the stack do not encounter this error

File: torch/_functorch/_aot_autograd/logging_utils.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+# TODO: It would be nice to reset the numbering every time aot_id goes
+# up, but this is annoying to do right now (because we don't know if
+# an aot_id will come back from the dead), so right now this also happens
+# to be a globally unique number too (at the cost of wobbling if you change
+# how the graphs compile)
+nth_graph: int = 0
+model_name: str = "model"
+
+
+def set_model_name(name):
```
Comment (bdhirsh) [inline torch/_functorch/_aot_autograd/logging_utils.py:20]: I see this isn't used anywhere - code-motion-only PR is great but maybe add a comment that we should just delete this later

Comment (jon-chuang) [reply torch/_functorch/_aot_autograd/logging_utils.py:20]: It's actually a public-facing API

Comment (jon-chuang) [reply torch/_functorch/_aot_autograd/logging_utils.py:20]: It's used elsewhere in torch

Comment (bdhirsh) [reply torch/_functorch/_aot_autograd/logging_utils.py:20]: yep you're right :( (used in dynamo)

**Review by bdhirsh — APPROVED** [review]

**Review by thiagocrepaldi — COMMENTED** [review]
@kit1980 do you know why the `release notes: onnx` label was applied here? I've been noticing this behavior in several PRs lately

**Pull request closed without merging**